WordPress/src/main/java/org/wordpress/android/ui/posts/adapters/AuthorSelectionAdapter.kt

```diff
@@ -0,0 +1,137 @@
+import org.wordpress.android.util.image.ImageManager
+import java.lang.ref.WeakReference
+import javax.inject.Inject
+
+class AuthorSelectionAdapter(context: Context) : SpinnerAdapter {
+    @Inject lateinit var imageManager: ImageManager
+    @Inject lateinit var uiHelpers: UiHelpers
+
+    private var observers: MutableList<WeakReference<DataSetObserver>> = mutableListOf()
+    private val items = mutableListOf<AuthorFilterListItemUIState>()
+
+    init {
+        (context.applicationContext as WordPress).component().inject(this)
+    }
+
+    override fun getCount(): Int = items.count()
+
+    override fun getDropDownView(position: Int, convertView: View?, parent: ViewGroup): View {
+        var view: View? = convertView
+        val holder: DropdownViewHolder
+
+        if (view == null) {
+            val inflater = LayoutInflater.from(parent.context)
+            view = inflater.inflate(R.layout.post_list_author_selection_dropdown, parent, false)
+            holder = DropdownViewHolder(view)
+            view.tag = holder
+        } else {
+            holder = view.tag as DropdownViewHolder
+        }
+
+        holder.bind(items[position], imageManager, uiHelpers)
+
+        return view!!
+    }
+
+    override fun getItem(position: Int): AuthorFilterListItemUIState = items[position]
+
+    override fun getItemId(position: Int): Long = position.toLong()
+
+    override fun getItemViewType(position: Int): Int = 0
+
+    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
+        var view: View? = convertView
+        val holder: NormalViewHolder
+
+        if (view == null) {
+            val inflater = LayoutInflater.from(parent.context)
+            view = inflater.inflate(R.layout.post_list_author_selection, parent, false)
+            holder = NormalViewHolder(view)
+            view.tag = holder
+        } else {
+            holder = view.tag as NormalViewHolder
+        }
+
+        holder.bind(items[position], imageManager, uiHelpers)
+
+        return view!!
+    }
+
+    override fun getViewTypeCount(): Int = 1
+    override fun hasStableIds(): Boolean = false
+    override fun isEmpty(): Boolean = false
+
+    fun notifyDataSetChanged() {
+        for (observer in observers) {
+            observer.get()?.onChanged()
+        }
+    }
+
+    override fun registerDataSetObserver(observer: DataSetObserver) {
+        for (alreadyAdded in observers) {
+            if (observer == alreadyAdded.get()) {
+                // No need to add since it is already added
+                return
+            }
+        }
+
+        observers.add(WeakReference(observer))
+    }
+
+    fun updateItems(newItems: List<AuthorFilterListItemUIState>) {
+        items.clear()
+        items.addAll(newItems)
+        notifyDataSetChanged()
+    }
+
+    override fun unregisterDataSetObserver(observer: DataSetObserver) {
+        for ((index, currentRef) in observers.withIndex()) {
```

Comment:
I think a similar suggestion to my comment in `registerDataSetObserver` would be applicable here. The changes will depend on what we do in the `registerDataSetObserver`, so I'll let you tackle this.

Reply:
Sure

Reply:
Refer to https://github.com/wordpress-mobile/WordPress-Android/pull/9484#discussion_r269376837